pages/instructorFileEditor/instructorFileEditor.js

```diff
@@ -932,7 +951,23 @@ function saveAndSync(fileEdit, locals, callback) {
                     } else if (result.hadJsonErrors) {
                         job.fail('One or more JSON files contained errors and were unable to be synced');
                     } else {
-                        job.succeed();
+                        if (config.chunksGenerator) {
```

Comment:
Nit/question, for my own understanding: the way I was thinking about things, all file edits should be served by servers that should unconditionally do chunk generation. So is this flag really needed? Or is it just for safety + explicitness?

Reply:
It's because some people might want to run a PL instance without using chunks at all, in a simple single-server configuration. The default is `chunksGenerator=false` and `chunksConsumer=false` so that things are easy for other people to set up (and we won't break existing installations at Berkeley, Maryland, etc). In the future we'll need to add admin docs about "how to set up a multi-server install".

Reply:
Gotcha! Related to this: I was thinking we'd want the default to actually be with chunks (at least for local dev) in the interest of trying to make things behave as similarly as possible between dev and prod. This is what we built the symlinking chunks thing for. Was that reconsidered?

Reply:
That's a great point. I think it's actually more that anything not on the critical path to deployment got ignored 😄.

Actually the `chunksGenerator`/`chunksConsumer` options were renamed when we were thinking about a different deployment strategy, which we discarded in the end. I think the justification above about single-server defaults is something of an after-the-fact justification that I just made up 😄.

Our current deployment strategy is to only opt-in courses one by one via the load balancer. That is, we only get the load balancer to redirect `/pl/course_instance/23423/*` to the chunk servers, for a couple of specific `course_instance_ids`. Initially we'll only redirect `pl-dev101` and `pl-test101`, and then we'll redirect the Calc 3 class that's running an exam on Tuesday. After we get through that we'll re-consider local dev.

I do agree that we want local dev to mimic the prod setup as much as possible.

We also don't have the test suite set up yet to use chunks.

Reply:
I've filed issues for the above items.